server/src/daemon/service.rs

```diff
@@ -48,6 +48,11 @@ impl Service {
         let (session, key) = Session::new(client_public_key, Arc::clone(&self.manager));
         // TODO: clean up the default generated key
         // TODO call self.manager.set_sessions();
+        self.manager
+            .lock()
+            .unwrap()
+            .insert_session(session.path().to_owned(), session.to_owned())
+            .await;
```

Comment:
Ran in to a new error here. Error: https://github.com/bilelmoussaoui/oo7/actions/runs/7987306810/job/21809515211?pr=73#step:4:369

Reply:
Looks like, I "can't hold the lock while you call .await. You have to release it before. If you want to hold if during an .await" (StackOverflow)[1]

[1]: [answer](https://stackoverflow.com/a/75080655) second comment.

Reply:
So, I think the only way to fix this is to remove `async` from `insert_session`. That means I have to remove `RwLock` [here](https://github.com/bilelmoussaoui/oo7/blob/01d1fa7d10c99f3cc9946e6144a0f6571619d619/server/src/daemon/service_manager.rs#L10). Could this be an issue?

Reply:
I think this won't be an issue, since `ServiceManager` is wrapped around `Mutex`.